ocp_resources/data_source.py

```diff
@@ -65,3 +65,31 @@ def pvc(self):
                 f"dataSource {self.name} is pointing to a non-existing PVC, name: {pvc_name}, "
                 f"namespace: {pvc_namespace}"
             )
+
+    def wait_for_status(self, status, timeout=TIMEOUT_4MINUTES, sleep=1):
+        """
+        Wait for data source to be in status
+
+        Args:
+            status: Expected status: True for a running VM, None for a stopped VM.
+            timeout (int): Time to wait for the resource.
+
+        Raises:
+            TimeoutExpiredError: If timeout reached.
+        """
+        LOGGER.info(
+            f"Wait for {self.kind} {self.name} status to be {'ready' if status == True else status}"
+        )
+        samples = TimeoutSampler(
```

Comment:
Please extract `TimeoutSampler` code to function and use it in both places. (with the logger)